### DISCUSSION TITLE
How can I limit the right-nav depth?

### DISCUSSION THREAD

**Discussion by cjw296** [pradyunsg/furo#557] [Questions] [answered]
For one of my projects, the right nav for the API reference has become unusable: it lists sections in the .rst file, then class names, then methods on those classes, all in the right nav.

What I'd like to do is limit the depth on the right name, so only the section heading and class names are shown.

How do I do that?

**Comment by pradyunsg** [comment]
Furo doesn't support any way to "hide" content from the right sidebar. It merely presents and styles the output from Sphinx for that.

See the `toc_*` configuration variables in Sphinx for some of the their toc controls: https://www.sphinx-doc.org/en/master/usage/configuration.html#confval-toc_object_entries

**Reply by cjw296** [reply]
`toc_object_entries` worked in this case but it's a shame it's a boolean rather than a depth: I would have preferred to include class names, but not methods within those classes.

The right-nav appears to be a layout choice that Furo has made, and it looks like there's [Furo code](https://github.com/pradyunsg/furo/blob/61045779cd46fc1e86cbe235de63f3b420f6cc43/src/furo/__init__.py#L122) that could take a depth?
Could Furo grow a configuration variable to set that to something other than -1?

**Reply by pradyunsg** [reply]
The code you're pointing to, is not related to this -- it's related to the left nav.

> toc_object_entries worked in this case but it's a shame it's a boolean rather than a depth: I would have preferred to include class names, but not methods within those classes.

That's a feature request you can make to Sphinx: https://github.com/sphinx-doc/sphinx/

**Reply by cjw296** [reply]
Apologies for getting the wrong code, would you be able to point me to the code that _is_ used to render the right nav? I suspect it'll help in correctly formulating the Sphinx feature request...

**Reply by pradyunsg** [reply]
https://github.com/pradyunsg/furo/blob/93e0093ce612a849f4751bc0506e1274e456816b/src/furo/theme/furo/page.html#L198

Uses the `toc` context variable provided directly by Sphinx.

**Reply by yuvaltassa** [reply]
@cjw296 we also want to do this.
Did you figure it out? 
Any helpful pointers?
Cheers

**Reply by cjw296** [reply]
@yuvaltassa - the code in Sphinx is surprisingly recent, only added last year I think.
I've got a feature request open [here](https://github.com/sphinx-doc/sphinx/issues/10935), probably best to subscribe to it and +1 my comment or some such.

**Reply by cjw296** [reply]
@pradyunsg - @mgeier suggests in https://github.com/sphinx-doc/sphinx/issues/10935#issuecomment-1324103568 that Furo may want to grow a `globaltoc_maxdepth` option, is that suitable for a feature request?

**Reply by pradyunsg** [reply]
No, that's unrelated. And, Furo isn't going to grow an option to enable hiding content based on depth from the left sidebar -- showing the entire site structure is an intentional design choice.

**Comment by pradyunsg** [comment]
FWIW, I've said this elsewhere: if you feel an urge to do this, the underlying problem might be that you have much content on the page.

https://github.com/pradyunsg/furo/discussions/584#discussioncomment-4184396 has the fuller comment.

**Reply by yuvaltassa** [reply]
@pradyunsg you are not wrong. We've taken your advice to heart and are looking into breaking up our too-long page.

But this request is different and would be relevant even after we did that. 

It is sometimes the case that you have lots of very small items that each deserves its own anchor, mostly so it can be refered to from elsewhere, but too small to deserve a TOC link.

As an example, consider our [XMLreference page](https://mujoco.readthedocs.io/en/latest/XMLreference.html). We are actually really happy with the way it is rendered at the moment, the right nav is a bit large but not unmanageably so. **However**, this is only because we are only referencing XML **elements** and not XML **attributes** (marked in red if you scroll down). We really want to add attribute links so we can link to them more precisely, but **we don't want these links in the right nav TOC**.

To sum up, I completely agree with your comment, and breaking up long pages is indeed good practice, but this does not  render invalid the request @cjw296 and I are making, that some levels of depth can, in some circumstances, be inappropriate for the TOC, even if they do have anchors.

Does that make sense?

**Reply by cjw296** [reply]
@pradyunsg - in my case, I've hit this only in the API reference of projects. These documents are intentionally TOC-dense and contain all the objects in the project; for this reason the right-nav is actually really important here, and having finer-grained control over what is and isn't included there would be really helpful.

**Comment by pradyunsg** [comment, marked as answer]
Adding a link to https://github.com/pradyunsg/furo/discussions/566#discussioncomment-4422331 for this.
